src/SparqlLua.php

```diff
@@ -4,27 +4,39 @@
 
 namespace ProfessionalWiki\SPARQL;
 
+use InvalidArgumentException;
+use MediaWiki\MediaWikiServices;
 use Scribunto_LuaLibraryBase;
 
 class SparqlLua extends Scribunto_LuaLibraryBase {
 
 	/**
-	 * @return mixed[]
+	 * @return array<mixed>
 	 */
 	public function register(): array {
 		return $this->getEngine()->registerInterface(
 			__DIR__ . '/Sparql.lua',
 			[
-				'runQuery' => fn( string $subjectId ): array => [ $this->runQuery( $subjectId ) ],
+				'runQuery' => fn ( string $subjectId ): array => [ $this->runSparqlQuery( $subjectId ) ],
 			]
 		);
 	}
 
 	/**
-	 * @return mixed[]
+	 * @param string $sparql
+	 *
+	 * @return array<mixed>
 	 */
-	private function runQuery( string $sparql ): array {
-		return [ $sparql ]; // TODO
+	private function runSparqlQuery( string $sparql ): array {
+		$sparqlEndpoint = MediaWikiServices::getInstance()->getMainConfig()->get( 'SPARQLEndpoint' );
+
+		if ( $sparqlEndpoint === false || !is_string( $sparqlEndpoint ) ) {
```

Comment:
The explicit boolean check is redundant, because that would be included in `!is_string()`.